xds/internal/balancer/outlierdetection/balancer.go

```diff
@@ -362,12 +361,11 @@ func (b *outlierDetectionBalancer) updateSubConnState(sc balancer.SubConn, state
 	b.scUpdateCh.Put(&scUpdate{
 		scw:   scw,
 		state: state,
-		cb:    cb,
 	})
 }
 
 func (b *outlierDetectionBalancer) UpdateSubConnState(sc balancer.SubConn, state balancer.SubConnState) {
-	b.updateSubConnState(sc, state, nil)
+	b.logger.Errorf("UpdateSubConnState(%v, %+v) called unexpectedly", sc, state)
```

Comment:
I don't get how if this logs an error and doesn't do anything do we keep it backward compatible?

Reply:
Why not just keep the old flow the same (UpdateSubConnState works normally), until we delete? What's the advantage of this?

Reply:
Oh, I guess this won't ever get called since the Channel now calls the listener callback.

Reply:
> Why not just keep the old flow the same (UpdateSubConnState works normally), until we delete? What's the advantage of this?

The goal is to catch all the places where we use the old API and update them to the new one so the old one can be deleted.  If we keep the old API code working then we might miss a place where we needed to convert it when we go to delete the API itself.